support-frontend/assets/pages/[countryGroupId]/components/__tests__/retryPaymentStatusTest.ts

```diff
@@ -0,0 +1,99 @@
+import type { StatusResponse } from 'helpers/forms/paymentIntegrations/readerRevenueApis';
+import { retryPaymentStatus } from '../retryPaymentStatus';
+
+describe('retryPaymentStatus', () => {
+	it('returns immediately if the payment is successful', async () => {
+		const status: StatusResponse = {
+			status: 'success',
+			trackingUri: 'https://support/status',
+		};
+		const getPaymentStatus = jest.fn(() => Promise.resolve(status));
+
+		const result = await retryPaymentStatus(getPaymentStatus, 1, 1);
```

Comment:
Is this actually testing anything? Won't passing in 1 for the maximum number of polls mean that the function will return immediately regardless of the result?

Reply:
I think it's testing the behaviour that if the promise immediately succeeds then no retries are necessary and the return value reflects the successful outcome.

Reply:
Yep, want to be assured it can succeed in first attempt